app/main/helpers/search_helpers.py

```diff
@@ -4,8 +4,9 @@
 from werkzeug.datastructures import MultiDict
 
 
-def get_lot_from_request(request):
-    return request.args.get('lot', None)
+def get_lot_from_request(request, all_lots):
+    lot = request.args.get('lot', None)
+    return lot if lot in all_lots or not all_lots else None
```

Comment:
I think this will this break if `all lots` is `None`, something like a "NoneType is not iterable" error.  It should be OK if you do the "not all_lots" check first, i.e.: `lot if not all_lots or lot in all_lots else None`